gneiss/plot/_radial.py

```diff
@@ -0,0 +1,144 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+import pandas as pd
+from gneiss.plot._dendrogram import UnrootedDendrogram
+
+try:
+    from bokeh.models.glyphs import Circle, Segment
+    from bokeh.models import ColumnDataSource, DataRange1d, Plot
+    from bokeh.models import HoverTool, BoxZoomTool, ResetTool
+except ImportError:
+    raise ImportWarning('Bokeh not installed. '
+                        '`radialplot` will not be available')
+
+
+def radialplot(tree, node_hue='node_hue', node_size='node_size',
+               node_alpha='node_alpha', edge_hue='edge_hue',
+               edge_alpha='edge_alpha', edge_width='edge_width',
+               figsize=(500, 500), **kwargs):
+    """ Plots unrooted radial tree.
+
+    Parameters
+    ----------
+    tree : instance of skbio.TreeNode
+       Input tree for plotting.
+    node_hue : str
+       Name of variable in `tree` to color nodes.
+    node_size : str
+       Name of variable in `tree` that species the radius of nodes.
+    node_alpha : str
+       Name of variable in `tree` to specify node transparency.
+    edge_hue : str
+       Name of variable in `tree` to color edges.
+    edge_alpha : str
+       Name of variable in `tree` to specify edge transparency.
+    edge_width : str
+       Name of variable in `tree` to specify edge width.
+    figsize : tuple, int
+       Size of resulting figure.  default: (500, 500)
+    **kwargs: dict
+       Plotting options to pass into bokeh.models.Plot
+
+    Returns
+    -------
+    bokeh.models.Plot
+       Interactive plotting instance.
+
+    Notes
+    -----
+    This assumes that the tree is strictly bifurcating.
+
+    See also
+    --------
+    bifurcate
+    """
+    # This entire function was motivated by
+    # http://chuckpr.github.io/blog/trees2.html
+    t = UnrootedDendrogram.from_tree(tree)
+
+    nodes = t.coords(figsize[0], figsize[1])
+
+    # fill in all of the node attributes
+    default_node_hue = '#D3D3D3'
+    nodes[node_hue] = pd.Series({n.name: getattr(n, node_hue,
+                                                 default_node_hue)
+                                 for n in t.levelorder(include_self=True)})
+
+    default_node_size = 1
```

Comment:
Is there a way to zip these two iterations into one? (I don't know I am just asking)